### PR TITLE
Limit treeinit iters with `limittreeinit_iters` and @error cleanly exit

### PR DESCRIPTION
**Pull request by Affie**
close https://github.com/JuliaRobotics/IncrementalInference.jl/issues/1035

### CODE REVIEW

**Comment by Affie** [comment]
I should maybe add test before we merge this. I'll see if I can get to it today.

**Comment by Affie** [comment]
Perhaps we should only increment init counter if `someInit=false`

File: test/testBasicTreeInit.jl [hunk 1]
```diff
@@ -71,6 +71,35 @@ tree, smt, hist = solveTree!(fg, timeout=70) # , verbose=true, verbosefid=verbos
...
+
+smtasks = Task[]
+
+solveTree!(fg; smtasks=smtasks, verbose=true)
+
+
+for var in good_vars
+    sppe = getVariable(fg,var) |> getPPE |> IIF.getPPESuggested
+    println("Testing ", var,": ", sppe)
+    @test isapprox(sppe[1], parse(Int,string(var)[end]), atol=0.1)
```
Comment (dehann) [inline test/testBasicTreeInit.jl:99]: should not be able to extract numerical result because there is no prior anywhere

**Review by dehann — CHANGES_REQUESTED** [review]

**Comment by dehann** [comment]
thanks the counting strategy sounds good

File: test/testBasicTreeInit.jl (hunk 1 above)
Comment (Affie) [reply test/testBasicTreeInit.jl:99]: The test tree looks like this:
image

one part solves fine and is tested numerically

Comment (Affie) [reply test/testBasicTreeInit.jl:99]: The test tree looks like this:
image
One part solves fine and has numerical results

**Comment by dehann** [comment]
It's topologically important to have a prior for gauge (absolute reference) rather than vague definition around one variable that happened to have 'initManual!' called.  That kind of information must be done via a Prior

**Comment by dehann** [comment]
besides the isapprox part think PR looks good thanks!

**Comment by Affie** [comment]
On `@errror` vs `@warn`
The current `ERROR` state propagation works with an @error and I didn't want the reason for it hidden as a `@warn`
Here is the output:
See Clique 2 starts the error and its propagated to Clique 4 that only exits because it received an error.
```julia
FSM cliq2, iter=76 -- postUpSolve_StateMachine, NO_INIT
┌ Error: CSM-2e Clique 2 tree init failed, max init retries reached.
└ @ IncrementalInference ~/build/JuliaRobotics/IncrementalInference.jl/src/CliqueStateMachine.jl:424
┌ Error: CSM clique 4: Exit with error state
└ @ IncrementalInference ~/build/JuliaRobotics/IncrementalInference.jl/src/CliqStateMachineUtils.jl:149
FSM cliq4, iter=88 -- exitStateMachine, NO_INIT┌ Error: CSM clique 2: Exit with error state
└ @ IncrementalInference ~/build/JuliaRobotics/IncrementalInference.jl/src/CliqStateMachineUtils.jl:149
FSM cliq2, iter=77 -- exitStateMachine, NO_INIT
[ Info: Finished tree based init-inference
Testing lm0: [0.01460663017623175]
Testing x0: [0.00541170226080372]
Testing x1: [1.0046086867937838]
Testing x2: [1.976521871242177]
Testing x3: [2.9868017889273593]
Test Summary:                                 | Pass  Total
basic tree initialization limittreeinit_iters |    5      5
```

**Pull request merged**